models/src/main/java/io/sphere/sdk/attributes/NestedType.java

```diff
@@ -0,0 +1,28 @@
+package io.sphere.sdk.attributes;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.core.type.TypeReference;
+import io.sphere.sdk.categories.Category;
+import io.sphere.sdk.channels.Channel;
+import io.sphere.sdk.models.Reference;
+import io.sphere.sdk.products.Product;
+import io.sphere.sdk.producttypes.ProductType;
+
+public class NestedType extends AttributeTypeBase {
+    private final Reference<ProductType> typeReference;
+
+    @JsonIgnore
+    private NestedType(final Reference<ProductType> typeReference) {
+        this.typeReference = typeReference;
+    }
+
+    public Reference<ProductType> getTypeReference() {
+        return typeReference;
+    }
+
+    @JsonCreator
+    public static NestedType of(final Reference<ProductType> typeReference) {
```

Comment:
Unfortunately, Jackson does not understand `Referenceable` so I still
needed to keep other method around.

On Tuesday, March 17, 2015, Laura Luiz <notifications@github.com> wrote:

> In models/src/main/java/io/sphere/sdk/attributes/NestedType.java
> <https://github.com/sphereio/sphere-jvm-sdk/pull/297#discussion_r26559515>
> :
>
> > +import io.sphere.sdk.producttypes.ProductType;
> > +
> > +public class NestedType extends AttributeTypeBase {
> > +    private final Reference<ProductType> typeReference;
> > +
> > +    @JsonIgnore
> > +    private NestedType(final Reference<ProductType> typeReference) {
> > +        this.typeReference = typeReference;
> > +    }
> > +
> > +    public Reference<ProductType> getTypeReference() {
> > +        return typeReference;
> > +    }
> > +
> > +    @JsonCreator
> > +    public static NestedType of(final Reference<ProductType> typeReference) {
>
> The one with Referenceable is below, this one has to be therefore removed,
> as it is unnecessary.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sphereio/sphere-jvm-sdk/pull/297/files#r26559515>.
>